From 98a7a7d2e7279954738103b4b8c9a927fea80fb6 Mon Sep 17 00:00:00 2001
From: Iurii Konovalenko <iurii.konovalenko@globallogic.com>
Date: Tue, 2 Aug 2016 19:31:13 +0300
Subject: [PATCH] arm64:dts:renesas: Adjust device tree for DomD on Salvator-x

Adjust device tree for DomD on Renesas Salvator-x board with Xen

Signed-off-by: Iurii Konovalenko <iurii.konovalenko@globallogic.com>
---
 .../boot/dts/renesas/r8a7795-salvator-x-domd.dts   | 257 +++++++--------------
 1 file changed, 80 insertions(+), 177 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-domd.dts b/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-domd.dts
index d153270..d0e85c2 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-domd.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-domd.dts
@@ -40,37 +40,10 @@
 	compatible = "renesas,salvator-x", "renesas,r8a7795";
 
 	aliases {
-		serial0 = &scif2;
 		serial1 = &scif1;
 		ethernet0 = &avb;
 	};
 
-	chosen {
-		bootargs = "ignore_loglevel rw root=/dev/nfs ip=dhcp";
-		stdout-path = "serial0:115200n8";
-	};
-
-	memory@48000000 {
-		device_type = "memory";
-		/* first 128MB is reserved for secure area. */
-		reg = <0x0 0x48000000 0x0 0x38000000>;
-	};
-
-	memory@500000000 {
-		device_type = "memory";
-		reg = <0x5 0x00000000 0x0 0x40000000>;
-	};
-
-	memory@600000000 {
-		device_type = "memory";
-		reg = <0x6 0x00000000 0x0 0x40000000>;
-	};
-
-	memory@700000000 {
-		device_type = "memory";
-		reg = <0x7 0x00000000 0x0 0x40000000>;
-	};
-
 	reserved-memory {
 		#address-cells = <2>;
 		#size-cells = <2>;
@@ -106,6 +79,53 @@
 		};
 	};
 
+	
+	soc {
+		interrupt-parent = <65000>;
+	};
+
+passthrough{
+
+	pass_extal_clk: extal {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		/* This value must be overridden by the board */
+		clock-frequency = <0>;
+	};
+
+	pass_extalr_clk: extalr {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		/* This value must be overridden by the board */
+		clock-frequency = <0>;
+	};
+
+	pass_audio_clk_a: audio_clk_a {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <0>;
+	};
+
+	pass_audio_clk_b: audio_clk_b {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <0>;
+	};
+
+	pass_audio_clk_c: audio_clk_c {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <0>;
+	};
+
+	/* External SCIF clock - to be overridden by boards that provide it */
+	pass_scif_clk: scif {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <0>;
+		status = "disabled";
+	};
+
 	mmngr {
 		compatible = "renesas,mmngr";
 	};
@@ -317,71 +337,7 @@
 		clock-frequency = <108000000>;
 	};
 
-	fixedregulator3v3: regulator@0 {
-		compatible = "regulator-fixed";
-		regulator-name = "fixed-3.3V";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-boot-on;
-		regulator-always-on;
-	};
-
-	fixedregulator1v8: regulator@1 {
-		compatible = "regulator-fixed";
-		regulator-name = "fixed-1.8V";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <1800000>;
-		regulator-boot-on;
-		regulator-always-on;
-	};
-
-	vcc_sdhi0: regulator@2 {
-		compatible = "regulator-fixed";
-
-		regulator-name = "SDHI0 Vcc";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-
-		gpio = <&gpio5 2 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
-
-	vccq_sdhi0: regulator@3 {
-		compatible = "regulator-gpio";
-
-		regulator-name = "SDHI0 VccQ";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <3300000>;
-
-		gpios = <&gpio5 1 GPIO_ACTIVE_HIGH>;
-		gpios-states = <1>;
-		states = <3300000 1
-			  1800000 0>;
-	};
-
-	vcc_sdhi3: regulator@4 {
-		compatible = "regulator-fixed";
-
-		regulator-name = "SDHI3 Vcc";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-
-		gpio = <&gpio3 15 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
-
-	vccq_sdhi3: regulator@5 {
-		compatible = "regulator-gpio";
-
-		regulator-name = "SDHI3 VccQ";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <3300000>;
-
-		gpios = <&gpio3 14 GPIO_ACTIVE_HIGH>;
-		gpios-states = <1>;
-		states = <3300000 1
-			  1800000 0>;
-	};
+};
 };
 
 &pwm1 {
@@ -424,10 +380,14 @@
 	};
 };
 
-&extal_clk {
+&pass_extal_clk {
 	clock-frequency = <16666666>;
 };
 
+&cpg {
+	clocks = <&pass_extal_clk>, <&pass_extalr_clk>;
+};
+
 &pfc {
 	pinctrl-0 = <&scif_clk_pins>;
 	pinctrl-names = "default";
@@ -445,10 +405,6 @@
 		renesas,groups = "scif1_data_a", "scif1_ctrl";
 		renesas,function = "scif1";
 	};
-	scif2_pins: scif2 {
-		renesas,groups = "scif2_data_a";
-		renesas,function = "scif2";
-	};
 	scif_clk_pins: scif_clk {
 		renesas,groups = "scif_clk_a";
 		renesas,function = "scif_clk";
@@ -518,36 +474,6 @@
 				 "audio_clkout_a", "audio_clkout3_a";
 		renesas,function = "audio_clk";
 	};
-
-	sdhi0_pins_3v3: sd0_3v3 {
-		renesas,groups = "sdhi0_data4", "sdhi0_ctrl";
-		renesas,function = "sdhi0";
-		power-source = <3300>;
-	};
-
-	sdhi0_pins_1v8: sd0_1v8 {
-		renesas,groups = "sdhi0_data4", "sdhi0_ctrl";
-		renesas,function = "sdhi0";
-		power-source = <1800>;
-	};
-
-	sdhi3_pins_3v3: sd3_3v3 {
-		renesas,groups = "sdhi3_data4", "sdhi3_ctrl";
-		renesas,function = "sdhi3";
-		power-source = <3300>;
-	};
-
-	sdhi3_pins_1v8: sd3_1v8 {
-		renesas,groups = "sdhi3_data4", "sdhi3_ctrl";
-		renesas,function = "sdhi3";
-		power-source = <1800>;
-	};
-
-	mmc0_pins_1v8: mmc0_1v8 {
-		renesas,groups = "sdhi2_data8", "sdhi2_ctrl";
-		renesas,function = "sdhi2";
-		power-source = <1800>;
-	};
 };
 
 &adsp {
@@ -559,17 +485,14 @@
 	pinctrl-0 = <&scif1_pins>;
 	pinctrl-names = "default";
 	/* Please use exclusively to the hscif1 node */
-	status = "okay";
-};
-
-&scif2 {
-	pinctrl-0 = <&scif2_pins>;
-	pinctrl-names = "default";
+	clocks = <&cpg CPG_MOD 206>,
+			 <&cpg CPG_CORE R8A7795_CLK_S3D1>,
+			 <&pass_scif_clk>;
 
 	status = "okay";
 };
 
-&scif_clk {
+&pass_scif_clk {
 	clock-frequency = <14745600>;
 	status = "okay";
 };
@@ -753,8 +676,8 @@
 		 <&cpg CPG_MOD 1028>, <&cpg CPG_MOD 1029>,
 		 <&cpg CPG_MOD 1030>, <&cpg CPG_MOD 1031>,
 		 <&cpg CPG_MOD 1019>, <&cpg CPG_MOD 1018>,
-		 <&audio_clk_a>, <&cs2000>,
-		 <&audio_clk_c>,
+		 <&pass_audio_clk_a>, <&cs2000>,
+		 <&pass_audio_clk_c>,
 		 <&cpg CPG_CORE R8A7795_CLK_S0D4>;
 
 	rcar_sound,dai {
@@ -773,12 +696,12 @@
 	shared-pin;
 };
 
-&wdt0 {
-	status = "okay";
+&pass_audio_clk_a {
+	clock-frequency = <22579200>;
 };
 
-&audio_clk_a {
-	clock-frequency = <22579200>;
+&gic{
+	compatible = "";
 };
 
 &i2c_dvfs {
@@ -940,43 +863,6 @@
 	};
 };
 
-&sdhi0 {
-	pinctrl-0 = <&sdhi0_pins_3v3>;
-	pinctrl-1 = <&sdhi0_pins_3v3>;
-	pinctrl-2 = <&sdhi0_pins_1v8>;
-	pinctrl-names = "default", "3v3", "1v8";
-
-	vmmc-supply = <&vcc_sdhi0>;
-	vqmmc-supply = <&vccq_sdhi0>;
-	cd-gpios = <&gpio3 12 GPIO_ACTIVE_LOW>;
-	wp-gpios = <&gpio3 13 GPIO_ACTIVE_HIGH>;
-	status = "okay";
-};
-
-&sdhi3 {
-	pinctrl-0 = <&sdhi3_pins_3v3>;
-	pinctrl-1 = <&sdhi3_pins_3v3>;
-	pinctrl-2 = <&sdhi3_pins_1v8>;
-	pinctrl-names = "default", "3v3", "1v8";
-
-	vmmc-supply = <&vcc_sdhi3>;
-	vqmmc-supply = <&vccq_sdhi3>;
-	cd-gpios = <&gpio4 15 GPIO_ACTIVE_LOW>;
-	wp-gpios = <&gpio4 16 GPIO_ACTIVE_HIGH>;
-	status = "okay";
-};
-
-&mmc0 {
-	pinctrl-0 = <&mmc0_pins_1v8>;
-	pinctrl-1 = <&mmc0_pins_1v8>;
-	pinctrl-names = "default", "1v8";
-
-	vmmc-supply = <&fixedregulator3v3>;
-	vqmmc-supply = <&fixedregulator1v8>;
-	bus-width = <8>;
-	status = "okay";
-};
-
 &csi2_40 {
 	status = "okay";
 
@@ -1076,3 +962,20 @@
 &vspi2 {
 	status = "okay";
 };
+
+&gpio3 {
+	status = "disabled";
+};
+
+&gpio4 {
+	status = "disabled";
+};
+
+&gpio5 {
+	status = "disabled";
+};
+
+&tsc3 {
+	status = "disabled";
+};
+
-- 
1.9.1

